sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -77,25 +90,12 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
                 throw new InvalidOperationException("Bearer token authentication is not permitted for non TLS protected (https) endpoints.");
             }
 
-            TokenRequestContext context;
-
-            // If the message already has a challenge response due to a sub-class pre-processing the request, get the context from the challenge.
-            if (message.HasResponse && message.Response.Status == (int)HttpStatusCode.Unauthorized && message.Response.Headers.Contains(HttpHeader.Names.WWWAuthenticate))
-            {
-                if (!TryGetTokenRequestContextFromChallenge(message, out context))
-                {
-                    // We were unsuccessful in handling the challenge, so bail out now.
-                    return;
-                }
-                Scopes = context.Scopes;
-            }
-            else
+            if (Scopes.Length > 0 && !message.Request.Headers.Contains(HttpHeader.Names.Authorization))
             {
-                context = new TokenRequestContext(Scopes, message.Request.ClientRequestId);
+               var context = new TokenRequestContext(Scopes, message.Request.ClientRequestId);
+               await AuthenticateRequestAsync(message, context, async).ConfigureAwait(false);
```

Comment:
I would expect this to be the base behavior of preprocess which a subclass for a client library could override. It seems that the key vault `ChallengeBasedAuthenticationPolicy` is doing this in `PreProcess`

Reply:
Originally I had this in the base `PreProcess`, but in discussing it with @pakrym, he raised the concern that sub-class implementers would need to always call `AuthenticateRequest` in overrides of `PrePRocess` to work properly, which may not be obvious. 

In the current implementation, the base has no opinion on whether sub-classes need to do anything with the Authentication header in `PreProcess`. If an Authorization header is added by sub-classes we will use it, else we will attempt to add one.

Reply:
As discussed earlier, I'll make this change along with the other suggested renames.